src/data.js

```diff
@@ -348,6 +348,63 @@ let Artifact = Entity.configure({
     'publicBucket',   // Public artifact bucket wrapping S3
     'monitor',        // base.monitor instance
   ],
+}).configure({
+  version: 2,
+  properties: {
+    taskId:         Entity.types.SlugId,
+    runId:          Entity.types.Number,
+    name:           Entity.types.String,
+    storageType:    Entity.types.String,
+    contentType:    Entity.types.String,
+    /**
+     * Location details storageType,
+     *
+     * storageType: s3
+     *   bucket:        S3 bucket that contains the object
+     *   prefix:        Prefix (path) for the object within the bucket
+     *
+     * storageType: azure
+     *   container:     Azure container that holds the blob
+     *   path:          Path to blob within container
+     *
+     * storageType: reference
+     *   url:           URL that artifact should redirect to
+     *
+     * storageType: error
+     *   reason:        Formalized reason for error artifact, see JSON schema.
+     *   message:       Human readable error message to return
+     * storageType: blob
+     *   contentType
+     *   contentEncoding
+     *   contentSha256: SHA256 hash of the un-encoded artifact
+     *   contentLength: Number of bytes of the un-encoded artifact
+     *   transferSha256: SHA256 hash of the content-encoding encoded artifact
+     *   transferLength: Number of bytes of the content-encoding encoded artifact
```

Comment:
Or maybe we should make `transferSha256 == contentSha256` if there is no difference... ie. let it default to `contentSha256` if not given... same for `transferLength`?